pkg/controller/oobpolicy.go

```diff
@@ -0,0 +1,135 @@
+	oobPolicyClient *oobpolicyclientset.Clientset) {
+	cont.initOOBPolicyInformerBase(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return oobPolicyClient.AciV1().OutOfBandPolicies(metav1.NamespaceAll).List(context.TODO(), options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return oobPolicyClient.AciV1().OutOfBandPolicies(metav1.NamespaceAll).Watch(context.TODO(), options)
+			},
+		})
+}
+
+func (cont *AciController) initOOBPolicyInformerBase(listWatch *cache.ListWatch) {
+	cont.oobPolicyInformer = cache.NewSharedIndexInformer(
+		listWatch,
+		&oobv1.OutOfBandPolicy{},
+		controller.NoResyncPeriodFunc(),
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.oobPolicyInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			cont.oobPolicyAdded(obj)
+		},
+		UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+			cont.oobPolicyUpdate(oldobj, newobj)
+		},
+		DeleteFunc: func(obj interface{}) {
+			cont.oobPolicyDelete(obj)
+		},
+	})
+}
+
+func (cont *AciController) oobPolicyAdded(obj interface{}) {
+	policy := obj.(*oobv1.OutOfBandPolicy)
+	cont.log.Infof("OutOfBandPolicy added: %s", policy.Name)
+	cont.updateOOBPolicy(policy)
+}
+
+func (cont *AciController) oobPolicyUpdate(oldobj interface{}, newobj interface{}) {
+	oldpolicy := oldobj.(*oobv1.OutOfBandPolicy)
+	newpolicy := newobj.(*oobv1.OutOfBandPolicy)
+	cont.log.Infof("OutOfBandPolicy updated: %s", newpolicy.Name)
+	if oldpolicy.Spec.VmmEpgDeploymentImmediacy == newpolicy.Spec.VmmEpgDeploymentImmediacy {
+		cont.log.Infof("OutOfBandPolicy update: No change in VmmEpgDeploymentImmediacy")
+		return
+	}
+	cont.updateOOBPolicy(newpolicy)
+}
+
+func (cont *AciController) oobPolicyDelete(obj interface{}) {
+	cont.indexMutex.Lock()
+	defer cont.indexMutex.Unlock()
+	policy := obj.(*oobv1.OutOfBandPolicy)
+	cont.log.Infof("OutOfBandPolicy deleted: %s", policy.Name)
+	fvRsDomAttSlice := cont.getfvRsDomAtt()
+	cont.deleteOOBPolicy(fvRsDomAttSlice)
+}
+
+func (cont *AciController) getfvRsDomAtt() []apicapi.ApicObject {
+	uri := fmt.Sprintf("/api/node/class/fvRsDomAtt.json?query-target-filter=and(wcard(fvRsDomAtt.dn,\"%s\"))", cont.config.AciPolicyTenant)
+	resp, err := cont.apicConn.GetApicResponse(uri)
+	if err != nil {
+		cont.log.Errorf("Failed to get response from APIC: %v", err)
+		return nil
+	}
+	return resp.Imdata
+}
+
+func (cont *AciController) postfvRsDomAtt(fvRsDomAttSlice []apicapi.ApicObject) {
+	uri := fmt.Sprintf("/api/node/mo/uni/tn-%s.json", cont.config.AciPolicyTenant)
+	err := cont.apicConn.PostApicObjects(uri, fvRsDomAttSlice)
+	if err != nil {
+		cont.log.Errorf("Failed to update the fvRsDomAtt: %v", err)
+	}
+}
+
+func (cont *AciController) updateFvRsDomAttInstrImedcy(fvRsDomAttSlice []apicapi.ApicObject, instrImedcy string) {
+	for _, fvRsDomAtt := range fvRsDomAttSlice {
```

Comment:
Only needs to be done for EPGs aci-containers-default and additional epgs that aci-containers creates by way of EPGAnnotation of namespaces

Reply:
Also set/reset reslmedcy attribute similarly.